src/tiled/tilesetdock.cpp

```diff
@@ -73,6 +76,32 @@ using namespace Tiled::Internal;
 
 namespace {
 
+class NoTilesetWidget : public QWidget
+{
+public:
+    explicit NoTilesetWidget(TilesetDock *parent = nullptr)
+        : QWidget(parent)
+    {
+        QGridLayout *gridLayout = new QGridLayout(this);
+
+        QSpacerItem *spacerVertTop = new QSpacerItem(1, 1, QSizePolicy::Ignored, QSizePolicy::Expanding);
+        gridLayout->addItem(spacerVertTop, 1, 0, Qt::AlignCenter);
```

Comment:
This isn't doing what you think it does. You're actually passing `Qt::AlignCenter` as the row-span argument to `addItem`, which explains why the centering isn't working correctly.

I found out that the use of spacers isn't needed at all though, since we can just pass `Qt::AlignCenter` when adding the New Tileset button. I'll push this change in a moment.